packages/react/src/hooks.ts

```diff
@@ -41,9 +41,26 @@ export type UseQueryData<T> =
 type QueryReturnType<Q> = Q extends Query<infer M> ? M : any;
 
 export function useQuery<Q extends Query<QueryReturnType<Q>>>(
-  on: UpdateType,
   queryProvider: () => Q,
   deps: any[],
+  on: UpdateType = UpdateType.ANY,
+): QueryReturnType<Q>[] {
+  const data = useQueryVerbose(queryProvider, deps, on);
+  if (data.status === 'error') {
+    throw data.error;
```

Comment:
Are these errors completely unexpected based on how the library works? Or is there ever a case where a user may want to gracefully handle an error in their component?

A more versatile usage might return the error as a member of a tuple or container object and allow the user to either render an error state, or rethrow it themselves if they use error boundary components.

Reply:
It could be a transient error when executing a query or a permanent error if the query is malformed for some reason. The latter being completely unexpected.

I'll mess around with returning `{ data, loading, error }`